numba/tests/test_asnumbatype.py

```diff
@@ -106,6 +145,57 @@ def test_nested_containers(self):
                 [types.Optional(self.int_nb_type), self.float_nb_type])),
         )
 
+    @unittest.skipIf(sys.version_info < (3, 9), "Requires Python 3.9")
```

Comment:
Introduce a `skip_unless_py39_or_later` in `numba.tests.support` and replace the decorator by:
```suggestion
    @skip_unless_py39_or_later
```